src/file_to_json.py

```diff
@@ -1,71 +1,94 @@
 import os
 import xlrd
 import pandas as pd
+from typing import Tuple, List, Dict
+from src.census_response import county_fips
+from src import data
 
 
-def file_to_json(input_dir, output_dir, blacklist=[]):
+def file_to_json(input_dir: str, output_dir: str, blacklist: List[str] = []) -> None:  # noqa: E501
     '''
     Reads excel/csv files into json format
     Excludes worksheets in blacklist
     '''
+    table_ls = files_to_dataframes(input_dir, blacklist)
+    dataframes_to_json_files(table_ls, output_dir)
+
+
+def file_to_wrapper(input_dir: str, blacklist: List[str] = []) -> data.Wrapper:  # noqa: E501
+    table_ls = files_to_dataframes(input_dir, blacklist)
+    combined_data = data.Wrapper()
+    for pair in table_ls:
+        combined_data.add(data.from_county_dataframe(pair[1]))
+    return combined_data
+
+
+def files_to_dataframes(input_dir: str, blacklist: List[str] = []) -> List[Tuple[str, pd.DataFrame]]:  # noqa: E501
+    table_ls: List[Tuple[str, pd.DataFrame]] = []  # unique name, DataFrame
+
     f_walk = os.walk(input_dir)
 
     for subdir, _, files in f_walk:
         for f in files:
-            # pandas not only provides file reading functionality,
-            # but outputs in a single format
-            # if performance requires, other packages may be implemented,
-            # that may require additional processing after reading in the file
             fp = os.path.join(subdir, f)
-            # using splitext allows '.' to appear in the filename
-            f_name, f_ext = os.path.splitext(f)
-            table_ls = []
-            if f_ext[:4] == '.xls':
-                # openpyxl can open .xlsx but not .xls
-                # xlrd can open .xls but not .xlsx
-                if f_ext == '.xlsx':
-                    engine = 'openpyxl'
+            table_ls = table_ls + file_to_dataframes(fp, blacklist)
+
+    normalized_tables = [(pair[0], normalize_dataframe(pair[1])) for pair in table_ls]  # noqa: E501
+
+    return normalized_tables
+
+
+def file_to_dataframes(filepath: str, blacklist: List[str] = []) -> List[Tuple[str, pd.DataFrame]]:  # noqa: E501
+    '''Returns list of (unique name, DataFrame)'''
+
+    directory, filename_with_extension = os.path.split(filepath)
+    f_name, f_ext = os.path.splitext(filename_with_extension)
+
+    if f_ext[:4] == '.xls':
```

Comment:
Maybe break this into its own function instead of this long nested if statement for Excel files?

May just look more difficult to read than it is with the removed code on github right below it.

Reply:
You were right. It was getting pretty far nested so I broke it into a separate function